### DISCUSSION TITLE
Related tools

### DISCUSSION THREAD

**Discussion by gusmanb** [gusmanb/logicanalyzer#30] [Show and tell]
If you have any project related to this and you want to share feel free to add a link in this discussion.

**Comment by gusmanb** [comment]
ESP32 frequency meter by [Gustavo Murta](https://github.com/Gustavomurta):

https://github.com/Gustavomurta/ESP32_frequencyMeter

**Reply by Gustavomurta** [reply]
Thanks a lot! 
As soon as possible I will translate all the project documentation that is in Portuguese.

**Comment by gusmanb** [comment]
Logic analyzer compatible with Sigrok by [pico-coder](https://github.com/pico-coder):

https://github.com/pico-coder/sigrok-pico

**Comment by Gustavomurta** [comment]
Hi Dr. Gusman, I also dream of a cheap and good scope for a long time. Decades actually. My first scope was mounted on a PC XT using a 12-bit ADC with a sample time of 3 us, in 90's decade. 
In my spare time, I play with these experiences.
I recently received some sample chips - high speed adcs. But I haven't had time to test them yet.

**MAX1193**  Ultra-Low-Power, 45Msps, Dual 8-Bit ADC  (for two channel scope) 
https://www.maximintegrated.com/en/products/analog/data-converters/analog-to-digital-converters/MAX1193.html

I'm an HW and not SW specialist. But I have studied and learned a lot. My current passion is with Golang.
https://go.dev/
Working with USB communication on various platforms.

Regards.

**Reply by gusmanb** [reply]
I will open a new discussion this evening about the plans for the oscilloscope board, feel free to join it :)

**Reply by extremesss10** [reply]
Having scope modules that could daisy chain with these would be an awesome addition.

**Reply by gusmanb** [reply]
Hi.

If you have an oscilloscope you can already chain it to the analyzer, connect the oscilloscope trigger to one of the EXT pins and use a pattern trigger on the analyzer.

Regarding the oscilloscope module, I have ordered two AD9283-100 ADC's and I'm waiting to receive them, once they arrive I will start testing if they're good for this :)

**Reply by gusmanb** [reply]
https://github.com/gusmanb/logicanalyzer/discussions/51

**Comment by Gustavomurta** [comment]
Hi Dr Gusmann, i found this module with AD9283:
https://numato.com/product/ad9283-adc-expansion-module/

And this guy, made some tests with ADC 100 Msps (MAX19506)  with FPGA (Tang nano 4K): 
https://www.youtube.com/results?search_query=btluni+fpga

**Reply by gusmanb** [reply]
It's shown as obsolete and there is no price, but I also found this module ( and it's cheap 😄 )

https://www.olimex.com/Products/FPGA/iCE40/iCE40-ADC/open-source-hardware

Also, there is a project that uses a pico and this module to create an oscilloscope, it's different of what I would like to do as I want to use a PC as viewer and add the triggering capabilities of the analyzer.

https://forum.lvgl.io/t/100-mhz-oscilloscope-with-raspberry-pi-pico-lvgl-v8-and-micropython/5669

Maybe an option could be to create a breakout board that divides the analyzer bus to two connectors for these modules, this for sure would speed up the development, I would need only to add the trigger logic and redo the communications, right now as the analyzer sends the data after capturing and parsing I use a very slow method with stdout, it's the easiest because you don't need to care at all about the USB side but for the oscilloscope this will need to be changed, and it may also give the option to create some new modes for the analyzer for lower speeds up to 10MSPS with infinite capture ammounts.

What do you think?

**Reply by Gustavomurta** [reply]
Hi Dr, 
This ADC08100 converter is relatively old (2000) and uses ladder switches, which is not the best type of converter. Just testing to be able to evaluate it.
The idea of preparing the logic analyzer for the use of ADC is very good. But it will have to add some features you mentioned.

**Reply by gusmanb** [reply]
Ok, then I will stick to the AD9283 ones. In any case, I think the ADC should not matter for the analyzer itself, we can design multiple modules so you can select which one to use, the base for the analyzer will be the same for all of them meanwhile they're parallel :)

**Comment by opcodex64** [comment]
At the moment I am working on interfacing a digital output MEMS accelerometer of type IIS3DBW using an ESP32S3 in context of a IoT project. Thanks to gusmanb and the community I could just integrate a RPI Pico as a LogicAnaylzer on the prototype PCB as shown in the attached picture. The low-cost of a RPI Pico and the free firmware/software makes this just possible.

Thanks to everyone who contributed to this nice project.

Kind regards,
Tugrul
![IMG_2433](https://github.com/user-attachments/assets/eedece24-50ab-4513-a697-b80e9422507a)
